### PR TITLE
feat: Added Pagination to the Resources Page

### PR DESCRIPTION
**Pull request by pdotsani**
## What type of PR is this? (check all applicable)

- [ ] ♻️ Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [x] 🎨 Enhancement
- [ ] 📝 Documentation Update
- [ ] 🔖 Release
- [ ] 🚩 Other

## Context

Closes issue #104 

## Screenshots/Recordings (if there are UI changes)

Screen Shot 2020-10-17 at 10 01 51 AM

## Implementation Details - what was your thought process as you changed the code?

- Modified query to toggle between pagination and search.
- Added Pagination component from material UI lab library.

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation (readme.md or contributing.md)?

- [ ] 📜 readme.md
- [ ] 📜 contributing.md
- [x] 🙅 no documentation needed
- [ ] 🙋 I'd like someone to help write documentation, and will file a new issue for it

### CODE REVIEW

**Comment by lpatmo** [comment]
YAY thank you for putting up this draft, @pdotsani! Do you need help getting the merge conflicts fixed? 

Separately, I'll tag this as `hacktoberfest-accepted` in case you'd like the credit.

**Comment by lpatmo** [comment]
Also, I won't review this until this is out of draft mode.

**Comment by pdotsani** [comment]
> YAY thank you for putting up this draft, @pdotsani! Do you need help getting the merge conflicts fixed?
> 
> Separately, I'll tag this as `hacktoberfest-accepted` in case you'd like the credit.

@lpatmo Looks like a rebase fixed all the conflicts. Should be good for a review.

File: src/components/Resources/Resources.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import React, { useState } from 'react';
```
Comment (lpatmo) [inline src/components/Resources/Resources.js:1]: I think this file can be removed in favor of `src/pages/Resources/Resources.js`, right?

File: src/pages/Resources/Resources.js [hunk 2]
```diff
@@ -4,15 +4,23 @@ import { useQuery } from 'react-query';
...
 import { getResources } from '../../utils/queries';
 
 function Resources() {
   const [searchValue, setSearchValue] = useState('');
-  const { isLoading, data, error } = useQuery(
-    [`?search=${searchValue}`],
-    getResources
-  );
+  const [goToPage, setGoToPage] = useState();
+  const [isPagination, triggerPagination] = useState();
```
Comment (lpatmo) [inline src/pages/Resources/Resources.js:14]: Was tempted to ask that this be renamed `paginate, setPaginate` for consistency's sake, but `triggerPagination` has a certain ring to it 🤔

File: src/pages/Resources/Resources.js [hunk 3]
```diff
@@ -58,6 +67,18 @@ function Resources() {
           </Typography>
         )}
         <br />
+        <Pagination
```
Comment (lpatmo) [inline src/pages/Resources/Resources.js:70]: Should we also hide the pagination UI when there is a `searchValue`? 

That way we won't see something like this: 
image

That "1" is clickable, leading to this UI which doesn't make sense: 
image

File: src/pages/Resources/Resources.js (hunk 2 above)
Comment (lpatmo) [reply src/pages/Resources/Resources.js:14]: Also, thinking we should initialize the value to be null: `useState();`

File: src/pages/Resources/Resources.js (hunk 3 above)
Comment (lpatmo) [inline src/pages/Resources/Resources.js:70]: UI nitpick: what do you think about moving the pagination component to the bottom of the grid? :)

**Comment by lpatmo** [comment]
🎉  Functionality works and you've figured out how to work with https://material-ui.com/components/pagination/, which is great!

Left a few comments re: minor details like:
- moving the component below the results 
- naming convention questions re: the isPagination, triggerPagination variable
- hiding the pagination component when search is in effect, so it won't be clickable -> cause confusion for users

Thanks @pdotsani! Happy to discuss more in threads if you want to push back or have questions

File: src/components/Resources/Resources.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  const [searchValue, setSearchValue] = useState('');
+  const [goToPage, setGoToPage] = useState();
+  const [isPagination, triggerPagination] = useState();
+  let params;
+
+  if (isPagination) {
+    params = `?page=${goToPage}`;
+  } else {
+    params = `?search=${searchValue}`;
+  }
```
Comment (turretd) [inline src/components/Resources/Resources.js:21]:
```suggestion
const params = isPagination? `?page=${goToPage}` : `?search=${searchValue}`;
```

File: src/pages/Resources/Resources.js [hunk 5]
```diff
@@ -4,15 +4,23 @@ import { useQuery } from 'react-query';
...
-  );
+  const [goToPage, setGoToPage] = useState();
+  const [isPagination, triggerPagination] = useState();
+  let params;
+
+  if (isPagination) {
+    params = `?page=${goToPage}`;
+  } else {
+    params = `?search=${searchValue}`;
+  }
```
Comment (turretd) [inline src/pages/Resources/Resources.js:21]:
```suggestion
const params = isPagination? `?page=${goToPage}` : `?search=${searchValue}`;
```

File: src/components/Resources/Resources.js (hunk 4 above)
Comment (lpatmo) [reply src/components/Resources/Resources.js:21]: Great suggestion, @turretd!

**Review by lpatmo — APPROVED** [review]

**Pull request merged**